### ISSUE TITLE
Ammonite REPL cannot run on Java 21

### ISSUE THREAD

**Issue by guofengzh** [com-lihaoyi/Ammonite#1380]
Because of the issue [ClassfileParser: allow missing param names (for JDK 21)](https://github.com/lampepfl/dotty/pull/17536), Ammonite REPL cannot run on Java 21.

Scala 3.3.1 has fixed its issue. It is kind to release a new Ammonite version that supports Scala 3.3.1 soon.

When runs Ammonite on Java 21, I got:
```
Loading...
Welcome to the Ammonite Repl 2.5.11 (Scala 3.2.2 Java 21)
@ error while loading ElementType,
class file java/lang/annotation/ElementType.class is broken, reading aborted with class java.lang.RuntimeException
bad constant pool index: 0 at pos: 1220
error while loading AccessFlag,
class file java/lang/reflect/AccessFlag.class is broken, reading aborted with class java.lang.RuntimeException
bad constant pool index: 0 at pos: 5189
```

**Comment by typingArtist** [comment]
The version 3.0.0-M0 release contains builds against Scala 3.3.1 from the 3.3-3.0.0-M0-53-084f7f4e build onwards. This and any following builds against Scala 3.3 work. This was released on Sep 13. The latest 2.5.11 release is from Sep 28 and doesn’t contain builds against Scala 3.3 as of now.

**Comment by lolgab** [comment]
Closing since this works on Ammonite versions published for Scala 3.3.1.
Install the latest with:
```
sh -c '(echo "#!/usr/bin/env sh" && curl -L https://github.com/com-lihaoyi/Ammonite/releases/download/3.0.0-M0/3.3-3.0.0-M0-59-cdeaa580) > /usr/local/bin/amm && chmod +x /usr/local/bin/amm' && amm
```
